SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,715 @@
+
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationDidBecomeActiveNotification object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationWillResignActiveNotification object:nil];
+
+    [self.primaryProtocolHandler stop];
+
+    self.streamingServiceTransportMap = [@{@(SDLServiceTypeAudio):@(SDLTransportClassInvalid),
+                                @(SDLServiceTypeVideo):@(SDLTransportClassInvalid)} mutableCopy];
+    self.secondaryTransportType = SDLSecondaryTransportTypeDisabled;
+    self.transportsForAudioService = @[];
+    self.transportsForVideoService = @[];
+
+    self.ipAddress = nil;
+    self.tcpPort = TCPPortUnspecified;
+}
+
+- (void)sdl_configureManager:(nullable NSArray<SDLSecondaryTransportTypeBox *> *)availableSecondaryTransports
+          availableTransportsForAudio:(nullable NSArray<SDLTransportClassBox *> *)availableTransportsForAudio
+          availableTransportsForVideo:(nullable NSArray<SDLTransportClassBox *> *)availableTransportsForVideo {
+    if (![self.stateMachine isCurrentState:SDLSecondaryTransportStateStarted]) {
+        SDLLogW(@"SecondaryTransportManager ignores duplicate Start Service ACK frame");
+        return;
+    }
+
+    SDLSecondaryTransportType secondaryTransportType = SDLSecondaryTransportTypeDisabled;
+    if (availableSecondaryTransports == nil || availableSecondaryTransports.count == 0) {
+        SDLLogW(@"Did not receive secondary transport type from system. Secondary transport is disabled.");
+        secondaryTransportType = SDLSecondaryTransportTypeDisabled;
+    } else {
+        // current proposal says the list should contain only one element
+        SDLSecondaryTransportTypeBox *transportType = availableSecondaryTransports[0];
+        secondaryTransportType = [transportType integerValue];
+    }
+
+    SDLSecondaryTransportType primaryTransportType = [self sdl_getTransportTypeFromProtocol:self.primaryProtocol];
+    if (primaryTransportType == secondaryTransportType) {
+        SDLLogW(@"Same transport is specified for both primary and secondary transport. Secondary transport is disabled.");
+        secondaryTransportType = SDLSecondaryTransportTypeDisabled;
+        // let audio and video services start on primary transport
+        availableTransportsForAudio = @[@(SDLTransportClassPrimary)];
+        availableTransportsForVideo = @[@(SDLTransportClassPrimary)];
+    } else if (secondaryTransportType == SDLSecondaryTransportTypeIAP) {
+        SDLLogW(@"Starting IAP as secondary transport, which does not usually happen");
+    }
+
+    // If SDL Core did not send transport list for the service, start it on Primary Transport.
+    if (availableTransportsForAudio == nil) {
+        SDLLogW(@"Transport for audio service is not specified by the system, using primary transport.");
+        availableTransportsForAudio = @[@(SDLTransportClassPrimary)];
+    }
+    if (availableTransportsForVideo == nil) {
```

Comment:
This `if` and the above are kind of redundant and can be combined, see above comment

Reply:
I updated the code in commit commit 068f8c142989fb8035fffed02f9c5d818b25d8cb. Hope the code is cleaner now.